Cabal/src/Distribution/Simple/GHC.hs

```diff
@@ -214,6 +215,29 @@ configure verbosity hcPath hcPkgPath conf0 = do
       progdb4 = Internal.configureToolchain implInfo ghcProg ghcInfoMap progdb3
   return (comp, compPlatform, progdb4)
 
+type GhcInfo = Map String String
+
+resloveGhcPath :: ConfiguredProgram -> GhcInfo -> FilePath
+resloveGhcPath ghcProg ghcInfo = fromMaybe (programPath ghcProg) realGhcPath
+  where
+    realGhcPath :: Maybe FilePath
+    --
+    -- This will give us the path to the actual `ghc` executable when `doctest`
+    -- is invoked via `cabal repl --with-ghc=doctest`.
+    --
+    -- Context:
+    --
+    -- If `doctest` is invoked via `cabal repl` it acts as a proxy for `ghc`.
+    -- It forwards all non-interactive invocations to `ghc` and only takes
+    -- responsibility when invoked with `--interactive`.  On `--info` it
+    -- augments the result of `ghc --info` with three additional fields:
+    --
+    -- 1. ghc: The path to the `ghc` executable that is used by `doctest`.
```

Comment:
Please note that even though I describe this in the context of `doctest`, other tools that make creative use of `cabal repl --with-ghc=..` (e.g. `hie-bios`) could benefit from this as well.  All they would need to do is augment `--info` with the actual path to `ghc`.